spin1_api/spin1_isr.c

```diff
@@ -556,14 +557,21 @@ INT_HANDLER soft_int_isr()
     // Clear software interrupt in the VIC
     vic[VIC_SOFT_CLR] = (1 << SOFTWARE_INT);
 
+    uint arg0  = user_event_queue.queue[user_event_queue.start].arg0;
+    uint arg1  = user_event_queue.queue[user_event_queue.start].arg1;
+
+    // Update queue pointer and trigger new event if queue not empty
+    user_event_queue.start = (user_event_queue.start + 1) % USER_EVENT_QUEUE_SIZE;
+
+    if (user_event_queue.start != user_event_queue.end) {
```

Comment:
Could save a couple of cycles by delaying the clearing of the interrupt and doing it only if the queue goes empty.